### PR TITLE
Fix map sub-layer popup component sizing

### PR DESCRIPTION
**Pull request by bsteinbk**
Use a different technique to register resize observer to ensure the controlled tree get proper size.

### CODE REVIEW

**Comment by mdastous-bentley** [comment]
Thanks for this fix Bill, 
any reason for not making useResizeObserver, useResizeDetector part of core-react?

**Comment by bsteinbk** [comment]
@mdastous-bentley - This resize observer is from itwinui-react, but it is not currently publicly available. @NancyMcCallB is going to work with that group to get it made public and when that happens we can remove it from the maplayers code.

File: extensions/map-layers/src/ui/widget/SubLayersTree.tsx [hunk 1]
```diff
@@ -12,14 +12,59 @@ import {
...
+* @internal
+ */
+function useResizeDetector(): { width: number | undefined, height: number | undefined, ref: React.Ref<HTMLDivElement> } {
+  const [width, setWidth] = React.useState<number>();
+  const [height, setHeight] = React.useState<number>();
+  const [ref] = useResizeObserver(React.useCallback((size: DOMRectReadOnly) => {
+    setWidth(size.width);
+    setHeight(size.height);
+    // eslint-disable-next-line no-console
+    console.log(`processing resize w=${size.width} h=${size.height}`);
```
Comment (mdastous-bentley) [inline extensions/map-layers/src/ui/widget/SubLayersTree.tsx:63]: remove ?

Comment (bsteinbk) [reply extensions/map-layers/src/ui/widget/SubLayersTree.tsx]: Removed console log.

**Review by mdastous-bentley — APPROVED** [review]

**Comment by mdastous-bentley** [comment]
@Mergifyio backport release/3.2.x

**Pull request merged**